scrapy/resolver.py

```diff
@@ -14,15 +16,23 @@ def __init__(self, reactor, cache_size, timeout):
         self.timeout = timeout
 
     def getHostByName(self, name, timeout=None):
-        if name in dnscache:
-            return defer.succeed(dnscache[name])
+        if PY2 and isinstance(name, text_type):
+            try:
+                safe_name = name.encode('idna')
+            except UnicodeError:
+                # we did our best, let it fail later.
+                safe_name = name
```

Comment:
Something changed with Twisted 17.1, where name is passed as Unicode (see https://github.com/scrapy/scrapy/issues/2728#issuecomment-300810555). I don't know if it's a bug or not on their end (they use getaddrinfo by default now, which is fine with Unicode names, so it may have been hidden)